src/api/js/src/high-level/high-level.ts

```diff
@@ -150,1724 +148,1807 @@ export function createApi(Z3: Z3Core): Z3HighLevel {
+      return isArith(obj) && isRealSort(obj.sort);
     }
 
-    isRealVal(obj: unknown): obj is RatNum {
+    function isRealVal(obj: unknown): obj is RatNum<Name> {
       const r = obj instanceof RatNumImpl;
-      r && this._assertContext(obj);
+      r && _assertContext(obj);
       return r;
     }
 
-    isRealSort(obj: unknown): boolean {
-      return this.isSort(obj) && obj.kind() === Z3_sort_kind.Z3_REAL_SORT;
+    function isRealSort(obj: unknown): boolean {
+      return isSort(obj) && obj.kind() === Z3_sort_kind.Z3_REAL_SORT;
     }
 
-    isBitVecSort(obj: unknown): obj is BitVecSort {
+    function isBitVecSort(obj: unknown): obj is BitVecSort<number, Name> {
       const r = obj instanceof BitVecSortImpl;
-      r && this._assertContext(obj);
+      r && _assertContext(obj);
       return r;
     }
 
-    isBitVec(obj: unknown): obj is BitVec {
+    function isBitVec(obj: unknown): obj is BitVec<number, Name> {
       const r = obj instanceof BitVecImpl;
-      r && this._assertContext(obj);
+      r && _assertContext(obj);
       return r;
     }
 
-    isBitVecVal(obj: unknown): obj is BitVecNum {
+    function isBitVecVal(obj: unknown): obj is BitVecNum<number, Name> {
       const r = obj instanceof BitVecNumImpl;
-      r && this._assertContext(obj);
+      r && _assertContext(obj);
       return r;
     }
 
-    isProbe(obj: unknown): obj is Probe {
+    function isProbe(obj: unknown): obj is Probe<Name> {
       const r = obj instanceof ProbeImpl;
-      r && this._assertContext(obj);
+      r && _assertContext(obj);
       return r;
     }
 
-    isTactic(obj: unknown): obj is Tactic {
+    function isTactic(obj: unknown): obj is Tactic<Name> {
       const r = obj instanceof TacticImpl;
-      r && this._assertContext(obj);
+      r && _assertContext(obj);
       return r;
     }
 
-    isAstVector(obj: unknown): obj is AstVector {
+    function isAstVector(obj: unknown): obj is AstVector<Name> {
       const r = obj instanceof AstVectorImpl;
-      r && this._assertContext(obj);
+      r && _assertContext(obj);
       return r;
     }
 
-    eqIdentity(a: Ast, b: Ast): boolean {
+    function eqIdentity(a: Ast<Name>, b: Ast<Name>): boolean {
       return a.eqIdentity(b);
     }
 
-    getVarIndex(obj: Expr): number {
-      assert(this.isVar(obj), 'Z3 bound variable expected');
-      return Z3.get_index_value(this.ptr, obj.ast);
+    function getVarIndex(obj: Expr<Name>): number {
+      assert(isVar(obj), 'Z3 bound variable expected');
+      return Z3.get_index_value(contextPtr, obj.ast);
     }
 
-    from(primitive: boolean): Bool;
-    from(primitive: number | CoercibleRational): RatNum;
-    from(primitive: bigint): IntNum;
-    from<T extends Expr>(expr: T): T;
-    from(expr: CoercibleToExpr): AnyExpr;
-    from(value: CoercibleToExpr): AnyExpr {
+    function from(primitive: boolean): Bool<Name>;
+    function from(primitive: number | CoercibleRational): RatNum<Name>;
```

Comment:
@bakkot This PR left out correct typing  for new `from` logic. `from(5)` will return `IntNum` but the overload here and in `types.ts` says that it will be `RatNum`

Reply:
@bakkot No worries. Added a small comment there. Cheers!